ohttp/src/lib.rs

```diff
@@ -177,18 +176,17 @@ impl KeyConfig {
     /// Not as a result of this function.
     pub fn encode(&self) -> Res<Vec<u8>> {
         let mut buf = Vec::new();
-        write_uint(size_of::<KeyId>(), self.key_id, &mut buf)?;
-        write_uint(2, u16::from(self.kem), &mut buf)?;
+        buf.write_u8(self.key_id)?;
+        buf.write_u16::<NetworkEndian>(u16::from(self.kem))?;
         let pk_buf = self.pk.key_data()?;
         buf.extend_from_slice(&pk_buf);
-        write_uint(
-            2,
-            u16::try_from(self.symmetric.len() * 4).unwrap(),
-            &mut buf,
-        )?;
+        if self.symmetric.len() > (u16::MAX / 4) as usize {
+            return Err(Error::Internal);
+        }
+        buf.write_u16::<NetworkEndian>((self.symmetric.len() * 4).try_into().unwrap())?;
```

Comment:
Previously this just crashed, which I think is totally OK in this case. If you want to do this, maybe instead implement `From<TryFromIntError>` for `Error`:

```rust
impl From<std::num::TryFromIntError> for Error {
  fn from(v: std::num::TryFromIntError) -> Self {
    Self::TooManySymmetricSuites
  }
}
```

Then all you need is:

```suggestion
        buf.write_u16::<NetworkEndian>((self.symmetric.len() * 4).try_into()?)?;
```

It's not an internal error per-se as it is down to having too many symmetric keys, so it might be better off with a new type, as above.

Reply:
This sounds great - thanks!